### PR TITLE
Added DeviceRendererFactory typing

### PR DESCRIPTION
**Pull request by OoDeLally**
## Description

Adds some typings for the device renderer.

## Type of change

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [X] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

-   [X] I've read & comply with the [contributing guidelines](https://github.com/Genymobile/genymotion-device-web-player/blob/main/CONTRIBUTING.md)
-   [X] I have tested my code for new features & regressions on both mobile & desktop devices, using the latest version of major browsers.
-   [X] I have made corresponding changes to the documentation (README.md).
-   [X] I've checked my modifications for any breaking changes.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  swipe?: KeyMap<KeyEffect & KeyEffectDistance>[];
+}
+
+type DeviceRendererKeyMapping = {
+  enable(isEnable: boolean): void;
+  setConfig(config: KeyMappingConfig): void;
+  activeKeyMappingDebug(isTraceActivate?: boolean, isGridActivate?: boolean): void;
+};
+
+type VmEvent = 'beforeunload' | 'fingerprint' | 'gps' | 'BATTERY_LEVEL' | string // TODO Provide an exhaustive list
```
Comment (OoDeLally) [inline index.d.ts:31]: TODO

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+type VmCommunication = {
+  disconnect(): void;
+  addEventListener(event: VmEvent, callback: (msg: string) => void): void; // TODO Verify if msg is always string
```
Comment (OoDeLally) [inline index.d.ts:35]: TODO

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+  sendData(data: { channel: string; messages: string[] }): void; // TODO Verify typing
```
Comment (OoDeLally) [inline index.d.ts:36]: TODO

File: package.json [hunk 4]
```diff
@@ -6,6 +6,7 @@
     "files": [
         "dist"
     ],
+    "types": "dist/index.d.ts",
```
Comment (OoDeLally) [inline package.json:9]: There is probably something to do with your bundler so it forwards the type file into the published npm package.

File: index.d.ts (hunk 2 above)
Comment (pgivel) [reply index.d.ts:35]: From what I've found in the code for now, all values for `msg` where strings

File: index.d.ts (hunk 1 above)
Comment (pgivel) [reply index.d.ts:31]: From what I've found, there are a few other events that are listed in the tests, but I don't know yet if the tests are exhaustive. Here are all values found in the tests:
```
ANDROID_ID
baseband
battery
BATTERY_STATUS
BLK
CLIPBOARD
diskio
fingerprint
framework
gps
IMEI
NETWORK
network_profile
settings
SYSTEM_PATCHER_LAST_RESULT
SYSTEM_PATCHER_STATUS
systempatcher
``` 
I'm trying to find a source of truth for these, either in the cloud player, maybe libcom repo, or something else.

File: index.d.ts [hunk 5]
```diff
@@ -28,12 +28,36 @@ type DeviceRendererKeyMapping = {
...
+  | 'BATTERY_LEVEL'
+  | 'BATTERY_LEVEL'
+  | 'BATTERY_STATUS'
+  | 'battery'
+  | 'beforeunload'
+  | 'BLK'
+  | 'CLIPBOARD'
+  | 'diskio'
+  | 'fingerprint'
+  | 'fingerprint'
```
Comment (jparez) [inline index.d.ts:43]: twice fingerprint

File: index.d.ts [hunk 6]
```diff
@@ -28,12 +28,36 @@ type DeviceRendererKeyMapping = {
...
+  | 'framework'
+  | 'gps'
+  | 'gps'
```
Comment (jparez) [inline index.d.ts:46]: twice gps

**Review by jparez — APPROVED** [review]

**Review by pgivel — APPROVED** [review]

**Pull request merged**